### ISSUE TITLE
DuckDB hang on `DELETE`

### ISSUE THREAD

**Issue by DerZc** [duckdb/duckdb#7530]
### What happens?

DuckDB hang on the following program:
```
CREATE TABLE t14(c0 BIGINT);
INSERT INTO t14(c0) VALUES ((1)), ((1)), ((1));
CREATE INDEX i1 ON t14(c0 );
DELETE FROM t14 WHERE t14.rowid;
```

### To Reproduce

I can reproduce this on the latest commit version https://github.com/duckdb/duckdb/commit/59a4ec3adce7e2bf67fe62912a8128bcbb6db0be

### OS:

ubuntu 22.04

### DuckDB Version:

https://github.com/duckdb/duckdb/commit/59a4ec3adce7e2bf67fe62912a8128bcbb6db0be

### DuckDB Client:

CLI

### Full Name:

Chi Zhang

### Affiliation:

Nanjing University, National University of Singapore

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by l1t1** [comment]
DELETE FROM t14 WHERE t14.rowid=1; also hang
if you delete from where c0 first, everything is fine

```
v0.7.2-dev3666 eae707d54c
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D CREATE TABLE t14(c0 BIGINT);
D INSERT INTO t14(c0) VALUES ((1)), ((1)), ((1));
D CREATE INDEX i1 ON t14(c0 );
D select * from t14;
┌───────┐
│  c0   │
│ int64 │
├───────┤
│     1 │
│     1 │
│     1 │
└───────┘
D select t14.*,rowid from t14;
┌───────┬───────┐
│  c0   │ rowid │
│ int64 │ int64 │
├───────┼───────┤
│     1 │     0 │
│     1 │     1 │
│     1 │     2 │
└───────┴───────┘
D delete from t14 where c0=0;
D delete from t14 where rowid=0;
D select t14.*,rowid from t14;
┌───────┬───────┐
│  c0   │ rowid │
│ int64 │ int64 │
├───────┼───────┤
│     1 │     1 │
│     1 │     2 │
└───────┴───────┘
D delete from t14 where t14.rowid=1;
D select t14.*,rowid from t14;
┌───────┬───────┐
│  c0   │ rowid │
│ int64 │ int64 │
├───────┼───────┤
│     1 │     2 │
└───────┴───────┘
```

**Comment by l1t1** [comment]
why delete from where c0 first,(even no data effected) everything is fine

**Comment by Mytherin** [comment]
Thanks for the report! Should be fixed by #7538

**Comment by DerZc** [comment]
Thanks for your work!
